src/main/java/com/google/api/codegen/transformer/java/JavaModelTypeNameConverter.java

```diff
@@ -0,0 +1,236 @@
+   * primitive, basicTypeName returns it in unboxed form.
+   */
+  private TypeName getTypeNameForElementType(TypeRef type, boolean shouldBoxPrimitives) {
+    String primitiveTypeName = PRIMITIVE_TYPE_MAP.get(type.getKind());
+    if (primitiveTypeName != null) {
+      if (primitiveTypeName.contains(".")) {
+        // Fully qualified type name, use regular type name resolver. Can skip boxing logic
+        // because those types are already boxed.
+        return typeNameConverter.getTypeName(primitiveTypeName);
+      } else {
+        if (shouldBoxPrimitives) {
+          return new TypeName(JavaTypeTable.getBoxedTypeName(primitiveTypeName));
+        } else {
+          return new TypeName(primitiveTypeName);
+        }
+      }
+    }
+    switch (type.getKind()) {
+      case TYPE_MESSAGE:
+        return getTypeName(type.getMessageType());
+      case TYPE_ENUM:
+        return getTypeName(type.getEnumType());
+      default:
+        throw new IllegalArgumentException("unknown type kind: " + type.getKind());
+    }
+  }
+
+  @Override
+  public String renderPrimitiveValue(TypeRef type, String value) {
+    Type primitiveType = type.getKind();
+    if (!PRIMITIVE_TYPE_MAP.containsKey(primitiveType)) {
+      throw new IllegalArgumentException(
+          "Initial values are only supported for primitive types, got type "
+              + type
+              + ", with value "
+              + value);
+    }
+    switch (primitiveType) {
+      case TYPE_BOOL:
+        return value.toLowerCase();
+      case TYPE_FLOAT:
+        return value + "F";
+      case TYPE_INT64:
+      case TYPE_UINT64:
+        return value + "L";
+      case TYPE_STRING:
+        return "\"" + value + "\"";
+      case TYPE_BYTES:
+        return "ByteString.copyFromUtf8(\"" + value + "\")";
+      default:
+        // Types that do not need to be modified (e.g. TYPE_INT32) are handled here
+        return value;
+    }
+  }
+
+  /**
+   * Returns the Java representation of a zero value for that type, to be used in code sample doc.
+   *
+   * Parametric types may use the diamond operator, since the return value will be used only in
+   * initialization.
+   */
+  @Override
+  public TypedValue getZeroValue(TypeRef type) {
+    // Don't call importAndGetShortestName; we don't need to import these.
+    if (type.isMap()) {
+      return TypedValue.create(typeNameConverter.getTypeName("java.util.HashMap"), "new %s<>()");
+    }
+    if (type.isRepeated()) {
+      return TypedValue.create(typeNameConverter.getTypeName("java.util.ArrayList"), "new %s<>()");
+    }
+    if (PRIMITIVE_ZERO_VALUE.containsKey(type.getKind())) {
+      return TypedValue.create(getTypeName(type), PRIMITIVE_ZERO_VALUE.get(type.getKind()));
+    }
+    if (type.isMessage()) {
+      return TypedValue.create(getTypeName(type), "%s.newBuilder().build()");
+    }
+    return TypedValue.create(new TypeName(""), "null");
```

Comment:
It may make no difference, but maybe this should pass in getTypeName(type) instead of an empty TypeName, similar to the PRIMITIVE_ZERO_VALUE case where it also makes no difference.